File: domain/src/main/java/com/ahn/domain/calculator/usecase/AddCalculatorHistoryUseCase.kt
package com.ahn.domain.calculator.usecase

import com.ahn.domain.calculator.model.CalculatorHistory
import com.ahn.domain.calculator.repository.CalculatorHistoryRepository
import javax.inject.Inject

class AddCalculatorHistoryUseCase @Inject constructor(
    private val repository: CalculatorHistoryRepository,
) {
    suspend operator fun invoke(history: CalculatorHistory) {
        repository.addHistory(history)
    }
}

File: ios/Packages/Domain/Sources/Domain/Calculator/UseCase/AddCalculatorHistoryUseCase.swift
public struct AddCalculatorHistoryUseCase: Sendable {
    private let repository: any CalculatorHistoryRepository

    public init(repository: any CalculatorHistoryRepository) {
        self.repository = repository
    }

    public func callAsFunction(_ history: CalculatorHistory) async throws {
        try await repository.addHistory(history)
    }
}
